tools/commentMonitor/client.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+type commentMonitorClient struct {
+	ghClient        *githubClient
+	allArgs         map[string]string
+	regex           *regexp.Regexp
+	eventMap        webhookEventMaps
+	eventType       string
+	commentTemplate string
+}
+
+// Validate comment if regexString provided.
+func (c *commentMonitorClient) validateRegex() error {
+	for _, e := range c.eventMap {
+		c.regex = regexp.MustCompile(e.RegexString)
+		if c.regex.MatchString(c.ghClient.commentBody) {
+			c.commentTemplate = e.CommentTemplate
+			c.eventType = e.EventType
```

Comment:
why do we need this variable?

Reply:
These are later used in `createRepositoryDispatch` call in the `extractArgs` function in this file.

Reply:
yep I sew how it is used now. thanks